src/index.js

```diff
@@ -20,8 +20,9 @@ ReactDOM.render(
 				}} />
 				<Route path="/grid/:entityId" component={GridView}/>
 				<Route path="/details/:parentId/:entityId" component={DetailView}/>
-			    <Route exact path="/settings" component={SettingsView}/>
-				<Route path="/settings/eventType/:eventTypeId/eventQuery/create" component={CreateEventQueryView}/>
+				<Route exact path="/settings" component={SettingsView}/>
+				<Route exact path="/settings/eventType/:eventTypeId/eventQuery/:isNewQuery/create" component={CreateEventQueryView}/>
+				<Route exact path="/settings/eventType/:eventTypeId/eventQuery/:eventQueryId/:isNewQuery/edit" component={CreateEventQueryView}/>
```

Comment:
We should consider using an understandable routing. `[...]/eventQuery/false/create` or `[...]/eventQuery/true/create` is not intuitive.
Why do we have to use `:isNewQuery` at all? In the _constructor_ you could check whether `:eventQueryId` is defined or not, couldn't you?